source/auth/tests/mongodb-oidc.rst

```diff
@@ -4,164 +4,397 @@ MongoDB OIDC
+  minute, and a refresh callback that returns data not conforming to
+  the ``OIDCRequestTokenResult`` with extra field(s).
+- Create a client with the callbacks.
+- Perform a ``find`` operation that succeeds.
+- Close the client.
+- Create a new client with the same callbacks.
+- Perform a ``find`` operation that fails.
+- Close the client.
 
-AWS Device Auth
-===============
+Cached Credentials
+==================
 
-Drivers MUST be able to authenticate using the "aws" device workflow simulating
-an EC2 instance with an enabled web identity token provider, generated by
-Drivers Evergreen Tools.
+Drivers MUST ensure that they are testing the ability to cache credentials.
+Unless otherwise specified, the tests MUST be performed with the callback-driven workflow with a provided request and refresh callback. If
+desired, the caching tests MAY be done using mock server responses.
+The following tests assume a global cache is in use.  If a different
+cache scheme is in use, appropriate tests MUST be written to ensure that
+the cache is performing as intended.
+
+Cache with refresh
+~~~~~~~~~~~~~~~~~~
+- Clear the cache.
+- Create a new client with a request callback that gives credentials that
+  expire in on minute.
+- Ensure that a ``find`` operation adds credentials to the cache.
+- Close the client.
+- Create a new client with the same request callback and a refresh callback.
+- Ensure that a ``find`` operation results in a call to the refresh callback.
+- Close the client.
+
+Cache with no refresh
+~~~~~~~~~~~~~~~~~~~~~
+- Clear the cache.
+- Create a new client with a request callback that gives credentials that
+  expire in one minute.
+- Ensure that a ``find`` operation adds credentials to the cache.
+- Close the client.
+- Create a new client with the a request callback but no refresh callback.
+- Ensure that a ``find`` operation results in a call to the request callback.
+- Close the client.
+
+Cache key includes callback
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+If the driver does not support using callback references or hashes as part of
+the cache key, skip this test.  This test ensures that the callback is
+considered as part of the cache key.
+
+- Clear the cache.
+- Create a new client with a request callback that does not give an
+  ```expiresInSeconds``` value.
+- Ensure that a ``find`` operation adds credentials to the cache.
+- Close the client.
+- Create a new client with a different request callback.
+- Ensure that a ``find`` operation adds a new entry to the cache.
+- Close the client.
+
+Error clears cache
+~~~~~~~~~~~~~~~~~~
+- Clear the cache.
+- Create a new client with a valid request callback that gives credentials
+  that expire within 5 minutes and a refresh callback that gives invalid
+  credentials.
+- Ensure that a ``find`` operation adds a new entry to the cache.
+- Ensure that a subsequent ``find`` operation results in an error.
+- Ensure that the cached token has been cleared.
```

Comment:
If a find causes a division by zero during query execution, is the intent to clear the cache?

Reply:
Not the full cache, but marking the access token as expired or removing it.

Reply:
(Could we use the term "Access Token" to refer to that specific token?)

Why would such an error indicate that the token is expired or otherwise invalid?

In cases where we don't have a refresh, the user would need to manually reauthenticate any time an error is encountered (even if it is otherwise ignored by the application). In cases with a refresh, it seems it would be called unnecessarily.